File: split-pane-layout-demo/src/main/java/com/mobidevelop/spl/demo/MainActivity.java
package com.mobidevelop.spl.demo;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import com.mobidevelop.spl.widget.SplitPaneLayout;

import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.util.Locale;

public class MainActivity extends Activity {
    private TextView first;
    private TextView second;
    private SplitPaneLayout layout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        first = findViewById(R.id.first);
        second = findViewById(R.id.second);

        layout = findViewById(R.id.layout);
        layout.setOnSplitterPositionChangedListener(new SplitPaneLayout.OnSplitterPositionChangedListener() {
            @Override
            public void onSplitterPositionChanged(SplitPaneLayout splitPaneLayout, boolean fromUser) {
                updateViews();
            }
        });

    }

    @Override
    protected void onResume() {
        super.onResume();
        updateViews();
    }

    private void updateViews() {
        NumberFormat percent = DecimalFormat.getPercentInstance(Locale.getDefault());

        first.setText(percent.format(layout.getSplitterPositionPercent()));
        second.setText(percent.format(1f - layout.getSplitterPositionPercent()));
    }
}

File: split-pane-layout-demo/src/main/java/com/mobidevelop/spl/demo/MainActivity.kt
package com.mobidevelop.spl.demo

import android.app.Activity
import android.os.Bundle
import com.mobidevelop.spl.widget.SplitPaneLayout
import com.mobidevelop.spl.widget.SplitPaneLayout.OnSplitterPositionChangedListener
import kotlinx.android.synthetic.main.activity_main.*
import java.text.DecimalFormat
import java.util.*

class MainActivity : Activity(), OnSplitterPositionChangedListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        layout.onSplitterPositionChangedListener = this
    }

    override fun onResume() {
        super.onResume()
        updateViews()
    }

    override fun onSplitterPositionChanged(splitPaneLayout: SplitPaneLayout, fromUser: Boolean) {
        updateViews()
    }

    private fun updateViews() {
        val percent = DecimalFormat.getPercentInstance(Locale.getDefault())
        first.text = percent.format(layout.splitterPositionPercent.toDouble())
        second.text = percent.format(1f - layout.splitterPositionPercent.toDouble())
    }
}